testing/checkFilterFits.py

```diff
@@ -0,0 +1,288 @@
+                    if not self.is_close(new_arr_bimol.Tmax.value_si, file_fit.Tmax.value_si):
+                        raise ValueError("""Arrhenius fits for reaction filtering need to be updated. Please run the
+                        ipython notebook {0} and commit the newly generated file.""".format(path_notebook))
+
+    def load_yaml(self, obj, path, class_dictionary):
+        """
+        Load Arrhenius fits for reaction filtering from YAML file.
+        """
+
+        with open(path, 'r') as f:
+            data = yaml.safe_load(stream=f)
+
+        data = {key: data[key] for key in data if key != 'class'}
+        args = recursive_make_object(data, class_dictionary)
+        obj.__init__(**args)
+
+    def analyze_reactions(self, Ts, fam_name, molecularity=1):
+        """
+        Refitting Arrhenius for reaction filtering using the current RMG database.
+        """
+        fam = self.database.kinetics.families[fam_name]
+        dep = fam.getTrainingDepository()
+        rxns = []
+
+        # Extract all training reactions for selected family
+        for entry in dep.entries.values():
+            r = entry.item
+            r.kinetics = entry.data
+            r.index = entry.index
+            for spc in r.reactants + r.products:
+                if spc.thermo is None:
+                    submit(spc)
+            rxns.append(r)
+
+        # Only proceed if at least one training reaction is available
+        if rxns:
+            # Get kinetic rates for unimolecular reactions
+            k_list = []
+            index_list = []
+            for rxn in rxns:
+                if len(rxn.reactants) == molecularity:
+                    k_list.append(rxn.kinetics)
+                    index_list.append(rxn.index)
+                if len(rxn.products) == molecularity:
+                    k_list.append(rxn.generateReverseRateCoefficient())
+                    index_list.append(rxn.index)
+
+            # Get max. kinetic rates at each discrete temperature
+            if k_list:
+                k_max_list = []
+                max_rxn_list = set()
+                for T in Ts:
+                    kvals = [k.getRateCoefficient(T) for k in k_list]
+                    mydict = dict(zip(index_list, kvals))
+
+                    # Find key and value of max rate coefficient
+                    key_max_rate = max(mydict.iteritems(), key=operator.itemgetter(1))[0]
+
+                    max_entry = dep.entries.get(key_max_rate)
+                    max_rxn = max_entry.item
+                    max_rxn_list.add(max_rxn)
+
+                    kval = mydict[key_max_rate]
+                    k_max_list.append(kval)
+
+                units = 's^-1' if molecularity == 1 else 'm^3/(mol*s)'
+
+                arr = Arrhenius().fitToData(Ts, np.array(k_max_list), units)
+                return arr
+
+        else:
+            arr = None
+            return arr
+
+    def is_close(self, a, b, rel_tol=1e-2, abs_tol=0.0):
```

Comment:
Make the following change to prevent this from being flagged by codacy:

```
    @staticmethod
    def is_close(a, b, rel_tol=1e-2, abs_tol=0.0):
```

Reply:
Thanks, done.